package PS;
import "proto/range.proto";

message DataConfig {
  enum DataFormat {
    BIN = 1;
    PROTO = 2;
    TEXT = 3;
  }
  required DataFormat format = 1;
  repeated string file = 2;
  // only valid for the binary format
  optional PbRange range = 3;
}

message ParameterInitConfig {
  enum Type {
    ZERO = 1;
    RANDOM = 2;
    FILE = 3;
  }
  optional Type type = 1 [default = ZERO];
  optional double random_std = 2 [default = 1];
}

message LearningRateConfig {
  enum Type {
    CONSTANT = 1;
  }
  required Type type = 1;
  optional double eta = 2 [default = 1];
}


message PenaltyConfig {
  enum Type {
    L1 = 1;  // lambda(0) * ||w||_1 + lambda(1) * ||w||_F^2
    L2 = 2;  // lambda(0) * ||w||_F^2
  }
  required Type type = 1;
  repeated double lambda = 2;
}
